lib/ansible/module_utils/vmware.py

```diff
@@ -799,6 +801,20 @@ def set_vm_power_state(content, vm, state, force):
     return result
 
 
+def wait_for_poweroff(vm, timeout=300):
+    result = dict()
+    interval = 15
+    while timeout:
+        if vm.runtime.powerState.lower() == 'poweredoff':
+            break
+        time.sleep(interval)
+        timeout -= interval
```

Comment:
This is going to end up in an endless loop if your timeout is not a multiple of 15.

```python
In [3]: interval = 5

In [4]: timeout = 13

In [5]: import time

In [6]: interval = 5

In [7]: timeout = 13

In [8]: while timeout:
   ...:     print timeout
   ...:     time.sleep(interval)
   ...:     timeout -= interval
   ...:
13
8
3
-2
-7
-12
-17
-22
-27
-32
^C```